magento/ece-tools
# MCLOUD-6135: Every step should throw only StepException

### Description

Every scenario step should throw only StepException.
If any error occurred on deploy phase - failed flag should be set and post-deploy skipped.

### Fixed Issues (if relevant)

1. https://jira.corp.magento.com/browse/MCLOUD-6135

### Manual testing scenarios

1. ssh to cloud environment and set read only permissions to file app/etc/env.php `chmod 444 app/etc/env.php`
2. run redeployment
Expected result:
 - deploy phase is failed and flag .deploy_failed is set. Post-deploy script is skipped.

### Release notes

For user-facing changes, add a meaningful release note. For examples, see [Magento Cloud ECE-tools release notes](https://devdocs.magento.com/cloud/release-notes/ece-release-notes.html).

### Associated documentation updates

Add link to Magento DevDocs PR or Issue, if needed.

### Contribution checklist
